x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/lib/alert_api.test.ts

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Alert, AlertType } from '../../types';
+import { httpServiceMock } from '../../../../../../../../src/core/public/mocks';
+import {
+  createAlert,
+  deleteAlerts,
+  disableAlerts,
+  enableAlerts,
+  loadAlerts,
+  loadAlertTypes,
+  muteAlerts,
+  unmuteAlerts,
+  updateAlert,
+} from './alert_api';
+
+const http = httpServiceMock.createStartContract();
+
+beforeEach(() => jest.resetAllMocks());
+
```

Comment:
It looks like we 're missing tests here around failures - http calls always fail :) we should ensure we handle this correctly for all of these API calls.
It also looks like some are made out of multiple calls (like `delete` which uses `Promise.all`), we should test for behaviour when some vs. all fail... even if we expect it to _just throw_ this should be tested and explicit in my opinion.